src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * Casts DateTimeInterface related classes to array representation.
+ *
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+class DateCaster
+{
+    public static function castDate(\DateTimeInterface $d, array $a, Stub $stub, $isNested, $filter)
```

Comment:
In fact, it casts all DateTimeInterface implementations. It's why I choosed the generic castDate method name. castDateTime seems only cast DateTime.